lib/Oscillator.js

```diff
@@ -1,12 +1,30 @@
 var Layer = require('./Layer');
 var audioNotes = require('audio-notes');
 var oscillatorParams = require('./mixins/oscillatorParams');
+var dsp;
+
+try {
+  dsp = require('../node_modules/dspjs/dsp.js');
+} catch(e) {
+  // dsp is not installed
+}
 
 var Oscillator = Layer.extend(oscillatorParams, {
 
   type: 'oscillator',
 
   _source: function (params) {
+
+    function validateAudioArray(arr){
```

Comment:
It's rarely a good idea to define functions inside other functions, and especially not in a performance critical path. This method will be called before each note is played, and a lot of extra garbage will be generated by having this function inline.

Let's move this and any other inline function to static class methods, and memoize them where it makes sense. For an example of how I've been doing memoization, check out the Params class.